### ISSUE TITLE
Release plan

### ISSUE THREAD

**Issue by lucaong** [njh/ruby-mqtt#143]
Hello, thanks a lot for the great gem :)

Is there a new released planned? The latest one is from 2017, and there were some useful fixes in the meanwhile. For example, the current release triggers a `warning: URI.unescape is obsolete `, that is fixed on `main`. Releasing a new version, even a patch one, would make it possible to upgrade and get rid of the annoying warning.

**Comment by rgaufman** [comment]
Is this gem still being maintained? - is there a better maintained MQTT gem out there?

**Comment by njh** [comment]
Making a release is currently blocked by #135 and merging in #139
But the build for #139 is currently broken.

**Comment by lucaong** [comment]
@njh the RuboCop issue should be easily fixed by changing `class OpenSSL::SSL::SSLSocket` to:

```ruby
module OpenSSL
  module SSL
    class SSLSocket
      # ...class body
    end
  end
end
```

Or, alternatively, by deactivating the RuboCop rule, which in my opinion does not contribute to readability/correctness, by adding this to `.rubocop.yml`:

```yaml
Style/ClassAndModuleChildren:
  Enabled: false
```

Probably the easiest way is for the maintainer to merge the PR, then fix the RuboCop issue as noted. Otherwise, if you prefer I can send a new PR.

**Comment by lucaong** [comment]
Hello @njh , sorry to bother again. I am sympathetic with the huge challenge of maintaining an open source project, so please do not take this as criticism.

I just think that cutting a new release, possibly after merging #146 to fix the RuboCop issue, could be a rather easy step, and yield a lot of value. Currently, version `5.0.0` of this gem cannot be used on latest Ruby 3.2 because of using the deprecated method `URI.unescape`, which does not exist anymore. That should be fixed on `main` already.

Please do let us know if something else is missing in order to cut a release, and me or someone else here can dedicate some time to it.

**Comment by njh** [comment]
Yes, would be really good to get a release out.

Annoyingly my Travis integration seems to have broken:
"Builds have been temporarily disabled for private and public repositories due to a negative credit balance. Please go to the [Plan page](https://app.travis-ci.com/account/plan) to replenish your credit balance."

I am not sure if there is something going on with my account or if Travis generally aren't supporting Open Source anymore.

**Comment by lucaong** [comment]
Yeah, that's annoying: Travis used to be my default choice for CI on open source projects, but since it was sold a few years back, things are not good anymore. On my open source projects, I now use the free tier of GitHub Actions. I can have a look if it's possible to easily migrate the current CI setup to GitHub Actions.

**Comment by lucaong** [comment]
I sent a PR to port the CI setup from Travis to GitHub Actions: https://github.com/njh/ruby-mqtt/pull/150

@njh as the maintainer, you probably have to enable the check on GitHub (and optionally require it to pass to merge a PR). On my fork, the actions run fine.

**Comment by njh** [comment]
Thank you very much for your help @lucaong.

I received the following reply from Travis CI:
> As you might already know, we have been making some changes to our pricing model [1] for the last year and this is also covered Free Trial Plan that is providing Travis CI for non-paid users. You can also consider them as OSS or education tier customers. 
> 
>  A free ‘Trial’ Plan, which any new user can select upon sign up, is a usage-based plan with an unlimited amount of users, which comes with a 10k trial pool of credits to start building right away. Once these credits run out, they are not replenished, and users must subscribe to a higher plan to continue building. Request [OSS Credits allowance](https://docs.travis-ci.com/user/billing-faq/#what-if-i-am-building-open-source), or please consider one of our available plans. Trial credits are valid only for 30 days.
> 
> While we are still working on the best fit for the open-source users to benefit from our platform, we do appreciate your commitment.

It implies that it is possible to get free credits for Open Source Projects, but it sounds like a pain and I don't like the uncertainty of if they will actually give them or not.

So moving to GitHub Actions makes sense, I think.
Will be great to get a new version of the gem released and get things into a better state.

**Comment by lucaong** [comment]
@njh thanks for merging #150 , I also just sent #151 to swap the obsolete Travis CI status badge on the README with the new GitHub Actions one.

**Comment by njh** [comment]
Wondering what the next release version number should be.

I have just reviewed the changelog between 0.5.0 and `main` and it is:
* Port CI setup from Travis to GitHub Actions
* Add Rubocop and performed code cleanup
* Updated yard minimum version to 
* Added support for clearing the queue of waiting incoming messages #117
* Wrap packet id after 0xffff (#118)
* Use a queue to wait for Puback packets rather than polling (#120)
* Filter out spec files from the coverage report
* Improvements `Packet.read_byte` performance (#134)
* Use monotonic clock instead of realtime for keep_alive (#132)
* Switch to CGI.unescape instead of the deprecated URI.unescape (#136)

Given that there is some new functionality, not just bug fixes, I think the next release number should be: `0.6.0`.

**Comment by lucaong** [comment]
It looks like there are no breaking changes, and the changes are a mixture of bug fixes (such as #136) and new features (such as #117). So yes, I agree that `0.6.0` is the appropriate version number.

**Comment by njh** [comment]
Version 0.6.0 is now released!

https://github.com/njh/ruby-mqtt/releases/tag/v0.6.0

Hopefully I won't need to rush out a 0.6.1 😬

Next, I hope to get this released:
https://github.com/njh/ruby-mqtt-packet

**Comment by njh** [comment]
@lucaong Thank you very much for your help in getting this release out

**Comment by lucaong** [comment]
Awesome! Thanks a lot @njh for making this happen!

This is a very useful gem, and it is great to have it running on Ruby 3 🎉
